libs/http/include/http/connection.hpp

```diff
@@ -55,18 +55,12 @@ class HTTPConnection : public AbstractHTTPConnection,
   }
 
   ~HTTPConnection() override
-  {
-    manager_.Leave(handle_);
-  }
+  {}
```

Comment:
The manager owns the connections, so no need to tell it that it is leaving, I think

Reply:
Connection is self-owning and it may well out-live the connection manager. This is what caused a problem to start with and what the PR #1885 fixes. If you look at how the connection is setup done in the server, you will see that the connection always passes shared self-pointers into lambdas that are passed to ASIO.

Reply:
It does do this yes, but I think there will be a race on destruction/socket closure - manager contains strong pointers to the connections in a map, and it lives in server.hpp so it is very possible they will destruct after asio is gone, causing bad stuff to happen.

I don't follow why the connections would need to out live the connection manager but happy to catch up whenever you are free.

Looking into the E2E failure - it seems that in both of our PRs there is some issue closing connections (E2E fails sending http calls) - I vaguely suspect that ```asio::async_read_until(socket_, *buffer_ptr, "\r\n\r\n", cb);``` might hang forever.